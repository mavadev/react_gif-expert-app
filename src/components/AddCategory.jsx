import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onAddCategory }) => {
	const [search, setSearch] = useState('');

	const onInputChange = ({ target }) => {
		setSearch(target.value);
	};

	const onSubmit = (event) => {
		event.preventDefault();
		if (search.trim() == '') return;

		onAddCategory(search.trim());
		setSearch('');
	};

	return (
		<form
			onSubmit={onSubmit}
			aria-label="form"
		>
			<input
				type="text"
				value={search}
				onChange={onInputChange}
				placeholder="Ej: Dragon Ball"
			/>
			<button>BUSCAR</button>
		</form>
	);
};

AddCategory.propTypes = {
	onAddCategory: PropTypes.func.isRequired,
};
