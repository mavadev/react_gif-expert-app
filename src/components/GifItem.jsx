import PropTypes from 'prop-types';

export const GifItem = ({ title, image, url, width, height }) => {
	return (
		<div
			className="card"
			data-testid="test_gif"
		>
			<a
				href={url}
				target="_blank"
				className="card-link"
				rel="noopener noreferrer"
			>
				<img
					src={image}
					alt={title}
					width={width}
					height={height}
				/>
				<div className="box-information">
					<p>{title}</p>
				</div>
			</a>
		</div>
	);
};

GifItem.propTypes = {
	title: PropTypes.string.isRequired,
	image: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
	width: PropTypes.string.isRequired,
	height: PropTypes.string.isRequired,
};
