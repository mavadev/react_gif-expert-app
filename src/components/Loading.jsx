import ReactLoading from 'react-loading';

export const Loading = () => {
	return (
		<div className="box-loading">
			<ReactLoading
				type="bars"
				color="black"
				width={125}
				height={125}
			/>
		</div>
	);
};
