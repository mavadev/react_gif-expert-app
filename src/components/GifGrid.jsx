import PropTypes from 'prop-types';
import { GifItem, Loading } from '../components';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
	const { gifs, isLoading } = useFetchGifs(category);

	return isLoading ? (
		<Loading />
	) : (
		<div
			className="box-category"
			data-testid="test_box-category"
		>
			<h3 data-testid="test_search-title">Resultados de "{category}"</h3>

			<div className="card-grid">
				{gifs.map((gif) => (
					<GifItem
						key={gif.id}
						{...gif}
					/>
				))}
			</div>
		</div>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
};
