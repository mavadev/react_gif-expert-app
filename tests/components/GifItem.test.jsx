import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {
	const GIF = {
		title: 'Saitama GIF',
		image: 'https://one-punch.com/saitama.jpg',
		url: 'https://mavadev.com',
		width: '300',
		height: '200',
	};

	test('debe hacer match con el snapshot', () => {
		const { container } = render(<GifItem {...GIF} />);
		expect(container).toMatchSnapshot();
	});

	test('debe mostrar el titulo del GIF', () => {
		render(<GifItem {...GIF} />);

		expect(screen.getByText(GIF.title)).toBeTruthy();
		// const { textContent } = screen.getByRole('paragraph');
		// expect(textContent).toBe(GIF.title);
	});

	test('debe mostrar la imagen con la URL y ALT indicado', () => {
		render(<GifItem {...GIF} />);

		const { src, alt } = screen.getByRole('img');
		expect(src).toBe(GIF.image);
		expect(alt).toBe(GIF.title);
	});

	test('debe mostrar la imagen con la URL y ALT indicado', () => {
		render(<GifItem {...GIF} />);

		const { src, alt } = screen.getByRole('img');
		expect(src).toBe(GIF.image);
		expect(alt).toBe(GIF.title);
	});

	test('debe redirigir a la URL original del GIF', () => {
		render(<GifItem {...GIF} />);

		const href = screen.getByRole('link').getAttribute('href');
		expect(href).toBe(GIF.url);
		// expect(screen.getByRole('link')).toHaveProperty('href', GIF.url + '/');
	});
});
