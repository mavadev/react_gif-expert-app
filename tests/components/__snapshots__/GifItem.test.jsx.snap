// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
    data-testid="test_gif"
  >
    <a
      class="card-link"
      href="https://mavadev.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Saitama GIF"
        height="200"
        src="https://one-punch.com/saitama.jpg"
        width="300"
      />
      <div
        class="box-information"
      >
        <p>
          Saitama GIF
        </p>
      </div>
    </a>
  </div>
</div>
`;
